lib/include/prjxray/xilinx/xc7series/ecc.h

```diff
@@ -0,0 +1,50 @@
+#ifndef PRJXRAY_LIB_XILINX_XC7SERIES_ECC_H_
+#define PRJXRAY_LIB_XILINX_XC7SERIES_ECC_H_
+
+#include <cstdint>
+
+namespace prjxray {
+namespace xilinx {
+namespace xc7series {
+
+// Extend the current ECC code with one data word (32 bit) at a given
+// index in the configuration packet and return the new ECC code.
+
+uint32_t icap_ecc(uint32_t idx, uint32_t data, uint32_t ecc) {
```

Comment:
Is there an advantage to doing this calculation one word at a time instead of taking a collection or a pair of iterators to do the calculation over?